### ISSUE TITLE
Some clients send out swapped out w/h with exif rotated images and thus css aspect-ratio has swapped out w/h

### ISSUE THREAD

**Issue by su-ex** [vector-im/element-web#18616]
### Steps to reproduce

As of https://github.com/matrix-org/matrix-react-sdk/commit/7b6536c297fbd44ab6cd1b66ce71d99c790852f7#diff-265bce8083cb791022167e130a59ebba99e3e974d9d3120f953ed41461b0f22a there's a problem with some exif rotated images due to the aspect-ratio and some clients sending swapped out w/h which makes css `aspect-ratio` have swapped w/h:

```
  "content": {
    "body": "left.jpg",
    "info": {
      "size": 3874,
      "mimetype": "image/jpeg",
      "w": 480,
      "h": 640,
      "xyz.amorgan.blurhash": "T3TI,axu~qofj[t7~qof00-;ayM{"
    },
    "msgtype": "m.image",
    "url": "mxc://supercable.onl/UIEMyFcdPHvVWaCpOScbCOuk"
  }
```

Swapped out w/h:
```
  "content": {
    "body": "left.jpg",
    "info": {
      "size": 3874,
      "mimetype": "image/jpeg",
      "w": 640,
      "h": 480,
      "xyz.amorgan.blurhash": "T3TI,axu~qofj[t7~qof00-;ayM{"
    },
    "msgtype": "m.image",
    "url": "mxc://supercable.onl/UIEMyFcdPHvVWaCpOScbCOuk"
  }
```

### What happened?

![Bildschirmfoto von 2021-08-18 12-35-19](https://user-images.githubusercontent.com/9221456/129884909-8bc8cd43-3f8c-4954-9b7e-20d0ffcc9a84.png)
![Bildschirmfoto von 2021-08-18 12-39-09](https://user-images.githubusercontent.com/9221456/129885004-5ceeaa51-8079-44c7-8462-896bcd5b925f.png)

### What did you expect?

Right `aspect-ratio`, at least after image has been loaded.

### Have you submitted a rageshake?

No

**Comment by dbkr** [comment]
I'm finding it a bit hard to tell but it looks like the image has been cropped - is that what's being demonstrated here? Probably we should letterbox it within the advertised container size?

**Comment by su-ex** [comment]
Before the mentionend commit it has been cropped, now it's size reduced and not well aligned.

**Comment by su-ex** [comment]
This is the timeline before the aforementioned commit:
![Bildschirmfoto von 2021-08-18 17-43-02_2](https://user-images.githubusercontent.com/9221456/129929324-d23d3243-91c4-4208-abed-14fbc0580b25.png)

**Comment by dbkr** [comment]
Reducing the size of the image seems to me to be a better way to handle this than cropping it though?

**Comment by su-ex** [comment]
Well, this doesn't change that it's misaligned.
This is fine while the image is not loaded, but it should be properly aligned when it is.

**Comment by dbkr** [comment]
OK - I've tried to update the description to reflect the problem, although still not really sure I've got it right. In particular with message bubbles - if your first screenshot is with message bubbles then it looks like the image is in the centre of the bubble which seems reasonable?

**Comment by su-ex** [comment]
Yes, "incorrect size metadata" probably fits best because it's likely still an issue when w and/or h are just plainly wrong instead of just swapped.
With the second part – I think I can't agree. It depends on the surrounding layout how it will look in the end.
I'd suggest "Images with incorrect (w/h) metadata should correct their alignment once the picture has actually been loaded" (I've avoided "size" as this is also a property of the metadata but can be ignored here).
